control/grpc.py

```diff
@@ -236,42 +260,92 @@ def __init__(self, config: GatewayConfig, gateway_state: GatewayStateHandler, rp
         self.subsys_max_ns = {}
         self.host_info = SubsystemHostAuth()
 
-    def create_host_psk_file(self, subsysnqn : str, hostnqn : str, psk_value : str) -> str:
+    def create_host_key_file(self, key_type : str, subsysnqn : str, hostnqn : str, key_value : str) -> str:
         assert subsysnqn, "Subsystem NQN can't be empty"
         assert hostnqn, "Host NQN can't be empty"
-        assert psk_value, "PSK value can't be empty"
+        assert key_value, "Key value can't be empty"
 
-        psk_dir = f"/tmp/psk/{subsysnqn}"
-        psk_file = f"{psk_dir}/{hostnqn}"
+        key_dir = f"/tmp/{key_type}/{subsysnqn}"
+        key_file = f"{key_dir}/{hostnqn}"
         try:
-            os.makedirs(psk_dir, 0o755, True)
+            os.makedirs(key_dir, 0o755, True)
         except Exception:
-            self.logger.exception(f"Error creating directory {psk_dir}")
+            self.logger.exception(f"Error creating directory {key_dir}")
             return None
         try:
-            with open(psk_file, 'wt') as f:
-                print(psk_value, end="", file=f)
-            os.chmod(psk_file, 0o600)
+            os.remove(key_file)
+        except FileNotFoundError:
+            pass
+        try:
+            with open(key_file, 'wt') as f:
+                print(key_value, end="", file=f)
```

Comment:
Nothing strictly wrong, but it's unusual to use `print` with files and removing auto-newline... when you already have the file `f` object:
```suggestion
                f.write(key_value)
```

Reply:
Done